src/matlab.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 The MathWorks, Inc.
+
+import * as core from "@actions/core";
+import * as http from "@actions/http-client";
+import * as io from "@actions/io";
+import * as tc from "@actions/tool-cache";
+import * as fs from "fs";
+import properties from "./properties.json";
+import * as script from "./script";
+
+export interface Version {
+    release: string;
+    semantic: string;
+}
+
+interface MATLABVersionInfo {
+    latest: string;
+    semantic: {
+        [release: string]: string | undefined
+    }
+}
+
+export async function toolcacheLocation(version: Version): Promise<string> {
+    let toolpath: string = tc.find("MATLAB", version.semantic);
+    if (toolpath) {
+        core.info(`Found MATLAB ${version.release} in cache at ${toolpath}`);
+    } else {
+        fs.writeFileSync(".keep", "");
+        toolpath = await tc.cacheFile(".keep", ".keep", "MATLAB", version.semantic);
+        io.rmRF(".keep");
+    }
+    return toolpath
+}
+
+export async function setupBatch(platform: string) {
+    const batchInstallDir = script.defaultInstallRoot(platform, "matlab-batch")
+    await script
+        .downloadAndRunScript(platform, properties.matlabBatchInstallerUrl, [batchInstallDir])
+        .then(() => {
+            core.addPath(batchInstallDir);
+        });
+    return
+}
+
+export async function getVersion(release: string): Promise<Version> {
+    const client: http.HttpClient = new http.HttpClient();
+    const versionInfo = await client.getJson<MATLABVersionInfo>(properties.matlabReleaseInfoUrl);
+
+    if (!versionInfo.result) {
+        return Promise.reject(Error(`Could not retrieve version info. Contact ci@mathworks.com if this problem persists.`));
+    }
+
+    let parsedRelease: string = release.toLowerCase();
+    if (parsedRelease === "latest") {
+        parsedRelease = versionInfo.result.latest;
+    }
+
+    let parsedSemantic = versionInfo.result.semantic[parsedRelease];
+    if (!parsedSemantic) {
+        return Promise.reject(Error(`Could not find version corresponding to specified release ${release}.`));
```

Comment:
@mw-hrastega

Reply:
I don't understand why we are using version and release in the same sentence.

Version of what? All the specified products?

Reply:
Because of how Github Actions operates, we need to be able to map each MATLAB release to a semantic version. For example "R2022b" = "9.13.0", "R2022a" = "9.12.0", and so on.

Reply:
Ok, so cannot find the product with that version, not the version itself. Right?

Reply:
Ok you can say:

`Unable to find the version corresponding to release ${release}.`

Reply:
I just realized that, for some reason, this is not showing the current code. This is the error that will show up if the user enters something in the `release` parameter that is invalid or out of our supported release range. I think `Specified release ${release} is invalid or not supported. Specify a valid release R2020a or later.` Or something similar could be better.

Reply:
updated